x-pack/plugins/uptime/public/components/overview/alerts/alerts_containers/alert_tls.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useDispatch, useSelector } from 'react-redux';
+import React, { useCallback } from 'react';
+import { AlertTlsComponent } from '../alert_tls';
+import { setAlertFlyoutVisible } from '../../../../state/actions';
+import { selectDynamicSettings } from '../../../../state/selectors';
+
+export const AlertTls = () => {
+  const dispatch = useDispatch();
+  const setFlyoutVisible = useCallback((value: boolean) => dispatch(setAlertFlyoutVisible(value)), [
+    dispatch,
+  ]);
+  const { settings } = useSelector(selectDynamicSettings);
+  return (
+    <AlertTlsComponent
+      ageThreshold={settings?.certThresholds?.age}
+      expirationThreshold={settings?.certThresholds?.expiration}
+      setAlertFlyoutVisible={setFlyoutVisible}
+    />
+  );
```

Comment:
i have been lately thinking, should we start using hooks for this pattern instead of container approach? I mean write a simple hook, which will dispatch action and return data, WDYT?

Reply:
It's a good idea, and I think we should do it in the future; however, we are going to have some interesting state management concerns for these alert creation UIs post-FF because they don't currently work with the Edit feature on the management page (because our store isn't available there). This only came to my attention after we finished NP migration.

For the time being I think we should keep these presentational components relying solely on props, until we devise a solution with the alerting team.

Reply:
Yes, it doesnt apply to this PR.